test/framework/src/main/java/org/elasticsearch/transport/AbstractSimpleTransportTestCase.java

```diff
@@ -1047,10 +1047,9 @@ public String executor() {
             .build());
 
         MockLogAppender appender = new MockLogAppender();
+        appender.start();
         Loggers.addAppender(LogManager.getLogger("org.elasticsearch.transport.TransportService.tracer"), appender);
```

Comment:
Instead maybe to move this line into the try/finally after starting the appender?